portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,43 +42,42 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String defaultCommentLimit = "5";
-    int commentLimit = Integer.parseInt(getParameter(request, "numShown", defaultCommentLimit));
+    int commentLimit = getIntInpOrDefault(request, "commentLimit", defaultCommentLimit);
 
     String defaultPageNumber = "0";
-    int pageNumber = Integer.parseInt(getParameter(request, "pageNumber", defaultPageNumber));
+    int pageNumber = getIntInpOrDefault(request, "pageNumber", defaultPageNumber);
 
-    String defaultSortMethod = "DESCENDING";
-    String sortMethod = getParameter(request, "sortMethod", defaultSortMethod);
+    String defaultSortMethod = "true";
+    String sortMethod = getParameterOrDefault(request, "sortMethod", defaultSortMethod);
 
     Query query = new Query("Comment");
     
-    if (sortMethod.equals(defaultSortMethod)) {
+    if (Boolean.parseBoolean(sortMethod)) {
       query.addSort("timestamp", SortDirection.DESCENDING);
     } else {
       query.addSort("timestamp", SortDirection.ASCENDING);
     }
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
     
-
-    ArrayList<Comment> comments = new ArrayList<>();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
     int startingIndex = pageNumber * commentLimit;
-    int count = 0;
-
-    for (Entity entity : results.asIterable()) {
-      if (count >= startingIndex) {
-        long id = entity.getKey().getId();
-        String comment = (String) entity.getProperty("comment");
-        long timestamp = (long) entity.getProperty("timestamp");
-
-        Comment newComment = new Comment(id, comment, timestamp);
-        comments.add(newComment);
-      }
-      count++;
-      if(count == startingIndex + commentLimit) {
-        break;
-      }
+
+    List<Entity> resultsList = datastore.prepare(query)
+                                        .asList(FetchOptions.Builder
+                                        .withLimit(startingIndex + commentLimit));
+
+    int endingIndex = Math.min(startingIndex + commentLimit, resultsList.size());
```

Comment:
what if I decide to troll you by putting ?commentLimit=-1 in my request

Reply:
their is a javascript client function isValidNumInp that checks that commentLimit is strictly positive, and if not, sets commentLimit to a default value. Is it best practice to extend a function like this to the java code as well?

Reply:
I have added an additional java method that extends this logic